File: programmers/python 3/12909.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12909

def solution(s):
    count = 0
    for bracket in s:
        count += 1 if bracket == '(' else -1
        if count < 0:
            return False
    return count == 0

File: programmers/swift/12909.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12909

func solution(_ s:String) -> Bool {
    var count = 0
    for bracket in s where count >= 0 {
        count += bracket == "(" ? 1 : -1
    }

    return count == 0
}